pkg/appgateserver/synchronous.go

```diff
@@ -40,29 +40,39 @@ func (app *appGateServer) handleSynchronousRelay(
 		return ErrAppGateHandleRelay.Wrapf("getting supplier URL: %s", err)
 	}
 
-	requestBodyBz, err := io.ReadAll(request.Body)
+	// Serialize the request to be sent to the supplier as a RelayRequest.Payload
+	// which will include the url, request body, method, and headers.
+	requestBz, err := httpcodec.SerializeHTTPRequest(request)
 	if err != nil {
-		return ErrAppGateHandleRelay.Wrapf("reading request body: %s", err)
+		return ErrAppGateHandleRelay.Wrapf("serializing request: %s", err)
 	}
-	request.Body.Close()
 
-	relayResponse, err := app.sdk.SendRelay(
-		ctx,
-		supplierEndpoint,
-		requestBodyBz,
-		request.Method,
-		request.Header,
-	)
+	relayResponse, err := app.sdk.SendRelay(ctx, supplierEndpoint, requestBz)
 	if err != nil {
 		return err
 	}
 
+	// Deserialize the RelayResponse payload to get the serviceResponse that will
+	// be forwarded to the client.
+	serviceResponse, err := httpcodec.DeserializeHTTPResponse(relayResponse.Payload)
+	if err != nil {
+		return ErrAppGateHandleRelay.Wrapf("deserializing response: %s", err)
+	}
+
 	app.logger.Debug().
-		Str("relay_response_payload", string(relayResponse.Payload)).
+		Str("relay_response_payload", string(serviceResponse.Body)).
 		Msg("writing relay response payload")
 
-	// Reply with the RelayResponse payload.
-	if _, err := writer.Write(relayResponse.Payload); err != nil {
+	// Reply to the client with the service's response status code and headers.
+	writer.WriteHeader(serviceResponse.StatusCode)
```

Comment:
Do we need a TODO_BLOCKER to handle errors properly?

Reply:
In this codepath, there has been no error originating from the `AppGateServer` and whatever reply (error or success) coming from the `RelayMiner` is just forwarded to the requester.

Reply:
#PUC